File: EstruturasFlexiveis/OutrasEstruturas/Stack/Java/Node.java
package EstruturasFlexiveis.OutrasEstruturas.Stack.Java;

public class Node<T> {
    //=====PRIVATE=====//
    private Node<T> next;
    private T obj;

    //=====CONSTRUCTOR=====//
    public Node() {
        this(null);
    }

    public Node(T obj) {
        this(obj, null);
    }

    public Node(T obj, Node<T> next) {
        this.setObj(obj);
        this.setNext(next);
    }

    //=====GET=====//
    public T getObj() {
        return this.obj;
    }

    public Node<T> getNext() {
        return this.next;
    }

    //=====SET=====//
    public void setObj(T obj) {
        this.obj = obj;
    }

    public void setNext(Node<T> next) {
        this.next = next;
    }

}

File: EstruturasFlexiveis/OutrasEstruturas/Stack/C/Node.c

#include "header/Node.h"

//=====CONSTRUCTOR=====//
Node* constructNode() {
    Node* node = (Node*)malloc(sizeof(Node));
    clearNode(node);
    return node;
}

Node* constructNodeObj(int obj) {
    Node* node = (Node*)malloc(sizeof(Node));
    node->next = NULL;
    node->obj = obj;
    return node;
}

//=====INITIALIZER=====//
void clearNode(Node* const node) {
    node->next = NULL;
    node->obj = 0;
}
